### PR TITLE
change mapping creation

### PR DESCRIPTION
**Pull request by nicolasThal**

### CODE REVIEW

File: ElasticaBundle/Command/OrchestraCreateIndexCommand.php [hunk 1]
```diff
@@ -34,6 +34,13 @@ protected function execute(InputInterface $input, OutputInterface $output)
     {
         $indexName = $this->getContainer()->getParameter('open_orchestra_elastica.index.name');
         $index = $this->getContainer()->get('open_orchestra_elastica.client.elastica')->getIndex($indexName);
-        $index->create(array('index' => array('number_of_shards' => 2, 'number_of_replicas' => 1)));
+        $index->create(
+            array(
+                'index' =>array(
```
Comment (ngilain) [inline ElasticaBundle/Command/OrchestraCreateIndexCommand.php]: a little space after `=>`

**Comment by ngilain** [comment]
:+1: when travis build is ok

**Comment by nicolasThal** [comment]
@itkg-ngilain I have added some more commits to the pr (to fix the build...)

File: ElasticaBundle/Resources/config/schema_initializer.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    open_orchestra_elastica.schema_initializer.manager:
+        class: %open_orchestra_elastica.schema_initializer.manager.class%
+
+    open_orchestra_elastica.schema_initializer.content_type:
+        class: %open_orchestra_elastica.schema_initializer.content_type.class%
+        arguments:
+            - @open_orchestra_model.repository.content_type
+            - @open_orchestra_elastica.schema_generator.content_type
+        tags:
+            - { name: open_orchestra_elastica.schema_initializer.strategy }
```
Comment (glenouvel) [inline ElasticaBundle/Resources/config/schema_initializer.yml:15]: eof

File: ElasticaBundle/Resources/config/factory.yml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+parameters:
+    open_orchestra_elastica.factory.mapping.class: OpenOrchestra\Elastica\Factory\MappingFactory
+
+services:
+    open_orchestra_elastica.factory.mapping:
+        class: %open_orchestra_elastica.factory.mapping.class%
```
Comment (glenouvel) [inline ElasticaBundle/Resources/config/factory.yml:6]: eof

File: Elastica/SchemaGenerator/ContentTypeSchemaGenerator.php [hunk 4]
```diff
@@ -51,11 +55,15 @@ public function createMapping($contentType)
...
-                $mapping['attribute_' . $field->getFieldId()] = array('type' => $this->formMapper->map($field->getType()), 'include_in_all' => false);
-                $mapping['attribute_' . $field->getFieldId() . '_stringValue'] = array('type' => 'string', 'include_in_all' => false);
+                $mappingProperties['attribute_' . $field->getFieldId()] = array('type' => $this->formMapper->map($field->getType()), 'include_in_all' => false);
+                $mappingProperties['attribute_' . $field->getFieldId() . '_stringValue'] = array('type' => 'string', 'include_in_all' => true);
             }
         }
 
-        $type->setMapping($mapping);
+        $mapping = $this->mappingFactory->create($type);
+//        $mapping->setParam('index_analyzer', 'indexAnalyzer');
```
Comment (glenouvel) [inline Elastica/SchemaGenerator/ContentTypeSchemaGenerator.php]: commented code ??

File: ElasticaBundle/Command/OrchestraCreateSchemaCommand.php [hunk 5]
```diff
@@ -32,10 +32,6 @@ protected function configure()
      */
     protected function execute(InputInterface $input, OutputInterface $output)
```
Comment (ngilain) [inline ElasticaBundle/Command/OrchestraCreateSchemaCommand.php:33]: As this protected method is now only an alias, is it still required ?

File: ElasticaBundle/DependencyInjection/Compiler/SchemaInitializerCompilerPass.php [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    /**
+     * You can modify the container here before it is dumped to PHP code.
+     *
+     * @param ContainerBuilder $container
+     *
+     * @api
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $managerName = 'open_orchestra_elastica.schema_initializer..manager';
```
Comment (ngilain) [inline ElasticaBundle/DependencyInjection/Compiler/SchemaInitializerCompilerPass.php]: I think there is a problem with the double dot

File: ElasticaBundle/Command/OrchestraCreateSchemaCommand.php (hunk 5 above)
Comment (nicolasThal) [inline ElasticaBundle/Command/OrchestraCreateSchemaCommand.php:33]: I need this proxy method so I can launch it from the console command.

I don't think that you can bypass this

**Comment by nicolasThal** [comment]
@itkg-ngilain @glenouvel pr updated

File: ElasticaBundle/Command/OrchestraCreateSchemaCommand.php (hunk 5 above)
Comment (ngilain) [inline ElasticaBundle/Command/OrchestraCreateSchemaCommand.php:33]: Oups sorry, didn't get the context (a command console), my bad :x

File: Elastica/SchemaGenerator/ContentTypeSchemaGenerator.php [hunk 7]
```diff
@@ -51,11 +55,15 @@ public function createMapping($contentType)
...
+//        $mapping->setParam('search_analyzer', 'searchAnalyzer');
```
Comment (ngilain) [inline Elastica/SchemaGenerator/ContentTypeSchemaGenerator.php]: dead code here

**Comment by ngilain** [comment]
:+1: without the dead code

**Pull request merged**